library/std/src/sys/windows/ext/process.rs

```diff
@@ -106,4 +119,104 @@ impl CommandExt for process::Command {
         self.as_inner_mut().creation_flags(flags);
         self
     }
+
+    fn arg_ext(&mut self, arg: impl Arg) -> &mut process::Command {
+        self.as_inner_mut().arg_ext(arg);
+        self
+    }
+
+    fn args_ext(&mut self, args: impl IntoIterator<Item = impl Arg>) -> &mut process::Command {
+        for arg in args {
+            self.arg_ext(arg);
+        }
+        self
+    }
+}
+
+/// Traits for handling a sized command.
+// FIXME: This really should be somewhere else, since it will be duplicated for unix. sys_common? I have no idea.
+// The implementations should apply to unix, but describing it to the type system is another thing.
+#[unstable(feature = "command_sized", issue = "74549")]
+pub trait CommandSized: core::marker::Sized {
+    /// Possibly pass an argument.
+    fn maybe_arg(&mut self, arg: impl Arg) -> io::Result<&mut Self>;
+    /// Possibly pass many arguments.
+    // FIXME: I don't know how to return the half-eaten iterator!
+    fn maybe_args(
+        &mut self,
+        args: impl IntoIterator<Item = impl Arg>,
+    ) -> Result<&mut Self, (usize, io::Error)>;
+    /// Build many commands.
+    fn xargs<I, S, A>(program: S, args: I, before: Vec<A>, after: Vec<A>) -> io::Result<Vec<Self>>
+    where
+        I: IntoIterator<Item = A>,
+        S: AsRef<OsStr> + Copy,
+        A: Arg;
```

Comment:
What does this do that `maybe_args` doesn't?

Reply:
Like its namesake, `xargs` is supposed to create a new command to fit in additional parameters when there's too many of them. As a result it returns possibly many commands.

It can still fail from a single oversized parameter being impossible to fit.